lib/CMakeLists.txt

```diff
@@ -213,15 +221,31 @@ target_compile_definitions(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,COMPILE_DEFIN
 target_include_directories(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,INCLUDE_DIRECTORIES>)
 target_compile_options(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,COMPILE_OPTIONS>)
 
-# add target specific files
-if(${QUDA_TARGET_TYPE} STREQUAL "CUDA")
-  add_subdirectory(targets/cuda)
+# nvshmem enabled parts need CUDA_SEPARABLE_COMPILATION ...
```

Comment:
How does this dslash object separation affect the order of how things are built?

Reply:
Not a lot for Ninja. For make it creates a somewhat artificial barrier IIRC. Unfortunately we cannot set SEPARABLE_COMPILATION on individual source files, so we need to have an additional target.
I currently do not have a better idea how to handle that without applying it to all of QUDA.

Reply:
Ok, understood.  We'll just tell folks to use `ninja` for NVSHMEM for now.

Reply:
^ has this note been added to the wiki?

Reply:
That they should use Ninja?
No, this only affects the parallelization fo the compile. Ninja does a better job in this case than make, but make works. No need to confuse users. Ninjas will use ninja anyway.